src/http/common.cr

```diff
@@ -166,7 +165,12 @@ module HTTP
 
   # :nodoc:
   def self.content_length(headers)
-    headers["Content-Length"]?.try &.to_u64?
+    length_header = headers.get? "Content-Length"
+    return nil unless length_header
+    if length_header.size > 1 && length_header.uniq.size != 1
+      raise ArgumentError.new("Multiple different values found for content-length #{length_header}")
```

Comment:
Perhaps `"Multiple Content-Length headers received did not match: #{length_headers}"` would be a better message?